clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/config/KubernetesConfiguration.java

```diff
@@ -53,6 +56,19 @@ public KubernetesConfigurationProperties kubernetesConfigurationProperties() {
     return new KubernetesConfigurationProperties();
   }
 
+  @Bean
+  @RefreshScope
+  @ConfigurationProperties("kubernetes")
+  @ConditionalOnProperty(
+      prefix = "kubernetes",
+      name = "custom-property-binding-enabled",
+      havingValue = "false",
+      matchIfMissing = true)
+  @Primary
```

Comment:
Is there a reason to mark it `@Primary`? Doing that will prevent any plugin to override it, which may be locking extensibility for kubernetes account management.

Reply:
Since there are two bean definitions, one here (the default bean) and the [one here](https://github.com/spinnaker/clouddriver/blob/00d498ed9cad0f90a4e8a862ffae5ca49ae9fccb/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/config/KubernetesCustomBinderConfiguration.java#L43-L48), I was getting the error where it says could not auto-wire, multiple bean definitions found. The way around it, at least for me, was to designate one of these as the primary. If there's a better way to do it, then I'm all ears.

Reply:
According to the `@ConditionalOnProperty` annotations of both beans, they are mutually exclusive and only one should be created at any given time. What value of `kubernetes.custom-property-binding-enabled` did you have when you got the error? If it was `false`, then the beans from `KubernetesCustomBinderConfiguration` should not be created, maybe that annotation needs a syntax like this one?

Reply:
So I was getting this error with `kubernetes.custom-property-binding-enabled: true` which made me resort to using `@Primary`:
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of method kubernetesCredentialsLoader in com.netflix.spinnaker.config.KubernetesConfiguration required a single bean, but 2 were found:
	- kubernetesAccountProperties: defined in BeanDefinition defined in class path resource [com/netflix/spinnaker/config/KubernetesConfiguration.class]
	- kubernetesAccounts: defined in BeanDefinition defined in class path resource [com/netflix/spinnaker/config/KubernetesCustomBinderConfiguration.class]


Action:

Consider marking one of the beans as @Primary, updating the consumer to accept multiple beans, or using @Qualifier to identify the bean that should be consumed

```

It turns out I simply needed to name the bean methods the same (i.e. `kubernetesAccountProperties`) and that makes it work without the `@Primary` notation.